File: com.codemachine/others/Aditya Verma/Binary Search/RotatedArray.java
import java.util.*;
class RotatedArray{
    static int rotatedArray(int[] Array, int low, int high){
        if(low > high){
            return -1;
        }
        int mid = (low + high) / 2;
        int prev = (mid + Array.length - 1) % Array.length;
        int next = (mid + 1) % Array.length;
        if (Array[mid] < Array[prev] && Array[mid] < Array[next]){
            return mid;
        }else if (Array[mid] > Array[high]){
            return rotatedArray(Array, mid+1, high);
        }else{
            return rotatedArray(Array, low, mid-1);
        }
    }
    public static void main(String[] args){
        int[] Array = {1,2,3,4,5,6,7,8,9,10,11,12};
        int rotationStartAt = rotatedArray(Array, 0, Array.length-1);
        if (rotationStartAt <= 0){
            System.out.println("Array is perfectly sorted!!!");
        }else{
            System.out.println("Array rotated by : "+(Array.length - rotationStartAt)+" steps!!!");
        }
    }
}

File: Coder File/searching/Binary Search/RotatedArray.py
def RotatedArray(lst):
    low = 0
    high = len(lst)-1
    while low <= high:
        mid = low + (high-low)//2
        if lst[mid] < lst[mid-1] and lst[mid] < lst[mid+1]:
            return mid
        elif lst[mid] > lst[low]:
            low = mid+1
        else:
            high = mid -1

lst = [8,9,10,11,12,13,1,2,3,4,5,6,7]
index = RotatedArray(lst)
print(index)